### DISCUSSION TITLE
About Indentation

### DISCUSSION THREAD

**Discussion by miguel-leon** [eslint/eslint#14835] [Q&A]
Hello,

Politely, I regard forcing a developer to use either spaces or tabs as something dumb. That is the developer's choice to make.
Though the important consideration is consistency within files written by said developer.

So, how can I add a rule to allow both tabs and spaces but validate consistency within a file?
For example, either 4 spaces or 1 tab consistently in a file.

Also, why is there a `"no-tab"` rule apart from the `"indent"` rule? what happens if I chose `"indent": ["error", "tab"]` together with `"no-tab": "error"`? Is this contradiction by design?

I hope I can get I kind response. I couldn't find anything out there, including stackoverflow.

Thanks.

**Comment by nzakas** [comment]
> So, how can I add a rule to allow both tabs and spaces but validate consistency within a file?
For example, either 4 spaces or 1 tab consistently in a file.

You would need to create a custom rule for that. You can start by copying the existing indent rule and modify it to do what you want. 

> Also, why is there a "no-tab" rule apart from the "indent" rule? what happens if I chose "indent": ["error", "tab"] together with "no-tab": "error"? Is this contradiction by design?

no-tabs checks for tab characters anywhere in the file, not just indents. It also has an option to not check indents. no-tabs was added for JSCS compatibility. Most of them time you just want to use indent.

**Reply by miguel-leon** [reply]
> You would need to create a custom rule for that. You can start by copying the existing indent rule and modify it to do what you want.

So, by default, not possible. Either force developers to one or the other even if not their preference, or remove this checking altogether.

> no-tabs checks for tab characters anywhere in the file, not just indents. It also has an option to not check indents. no-tabs was added for JSCS compatibility. Most of them time you just want to use indent.

I understand now.

Well. Thanks for your answers. ✌️
